components/bl-pdfViewer-component/src/index.js

```diff
@@ -0,0 +1,105 @@
+import { Document, Page } from './react-pdf.min.js';
+import { useState, useEffect, useRef } from 'react';
+import { Controls, NoData } from './subcomponents';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function PdfViewer({ component, eventHandlers }) {
+  const { style, display, classList, pdfUrl, renderType, width, height } = component;
+  const { onLoadSuccess, onLoadError } = eventHandlers;
+
+  const [numPages, setNumPages] = useState(null);
+  const [pageIndex, setPageIndex] = useState(1);
+  const [documentRef, setDocumentRef] = useState();
+  const [controlsDisabled, setControlsDisabled] = useState(false);
+
+  const inputRef = useRef();
+
+  useEffect(() => {
+    inputRef.current.style.width = ((inputRef.current.value.length + 1) * 9) + 'px';
+  }, [`${ pageIndex }`.length]);
+
+  useEffect(() => {
+    if (documentRef) {
+      documentRef.style.height = height;
+      documentRef.style.width = width;
+    }
+  }, [documentRef]);
+
+  const onDocumentLoadSuccess = ({ numPages }) => {
+    setNumPages(numPages);
+    onLoadSuccess({ pageCount: numPages });
+  };
+
+  const onDocumentLoadError = (error) => {
+    onLoadError({ message: error.message });
+  };
+
+  const onPageLoadSuccess = () => {
+    documentRef.firstChild.firstChild.style.width = width;
```

Comment:
`firstChild.firstChild` - this approach fixes elements, you can't move or add a new one with it.

Check if the library provides `forwardRef`(preferred) for an element which you try to size or use query selectors/get by id